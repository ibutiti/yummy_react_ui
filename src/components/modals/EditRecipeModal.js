import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import RecipeForm from '../forms/RecipeForm';
import { Modal, Icon, Card } from 'semantic-ui-react';
import { editRecipe } from '../../actions/recipes';
import { fetchCategories } from '../../actions/categories';

class EditRecipeModal extends Component {
  componentDidMount() {
    this.props.fetchCategories(1, 1000, this.props.apiKey);
  }
  state = {
    modalOpen: false
  };

  modalClick = () => {
    this.setState({ modalOpen: !this.state.modalOpen });
  };

  submit = data =>
    this.props.editRecipe(data, this.props.id, this.props.apiKey).then(() => {
      this.setState({ modalOpen: false });
    });

  render() {
    return (
      <div>
        <Modal
          basic
          size="small"
          open={this.state.modalOpen}
          style={{ marginTop: '180px', marginLeft: '350px' }}
          trigger={<Icon onClick={this.modalClick} name="edit" size="small" />}>
          <Modal.Content>
            <Modal.Description>
              <Card color="violet" fluid>
                <Card.Content>
                  <Card.Header>Edit your Recipe</Card.Header>
                  <RecipeForm
                    title={this.props.title}
                    category_id={this.props.category_id}
                    content={this.props.content}
                    categories={this.props.categories}
                    submit={this.submit}
                    cancel={this.modalClick}
                  />
                </Card.Content>
              </Card>
            </Modal.Description>
          </Modal.Content>
        </Modal>
      </div>
    );
  }
}
EditRecipeModal.propTypes = {
  editRecipe: PropTypes.func.isRequired,
  apiKey: PropTypes.string.isRequired
};

function mapStateToProps(state) {
  return {
    apiKey: state.user.apikey,
    categories: state.categories.data
  };
}

export default connect(mapStateToProps, { editRecipe, fetchCategories })(
  EditRecipeModal
);
