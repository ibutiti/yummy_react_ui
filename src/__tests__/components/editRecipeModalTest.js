import React from 'react';
import Enzyme, { mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ReactDom from 'react-dom';
import configureMockStore from 'redux-mock-store';
import { MemoryRouter } from 'react-router-dom';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';

import EditRecipeModal from '../../components/modals/EditRecipeModal';

Enzyme.configure({ adapter: new Adapter() });

describe('<EditRecipeModal>', () => {
  const store = configureMockStore([thunk])({
    recipes: { links: {}, data: {} },
    categories: { links: {}, data: {} },
    user: { apikey: 'abcd' }
  });
  const props = {
    modalClick: jest.fn(),
    editRecipe: jest.fn(() => Promise.resolve('recipes')),
    fetchCategories: jest.fn(() => Promise.resolve('categories')),
    submit: jest.fn(),
    setState: jest.fn(),
    modalOpen: false
  };

  // test it renders without crashing
  it('should render without crashing', () => {
    const { enzymeWrapper } = mount(
      <Provider store={store}>
        <EditRecipeModal {...props} />
      </Provider>
    );
  });

  it('renders fully', () => {
    const tree = renderer
      .create(
        mount(
          <Provider store={store}>
            <EditRecipeModal {...props} />
          </Provider>
        )
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
