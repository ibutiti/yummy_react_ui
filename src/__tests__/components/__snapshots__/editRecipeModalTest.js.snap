// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditRecipeModal> renders fully 1`] = `
<div>
  <i
    aria-hidden="true"
    className="edit small icon"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  />
</div>
`;
